TrackModel/src/trackmodel/model/Block.java

```diff
@@ -407,27 +392,27 @@ public void setCumElevation(float newCumElevation) {
   }
 
   /**
-   * This method will set the new direction of a block.
-   * @param newDirection A integer value will be passed in to set the direction
+   * This method will set if the block is bidirectional.
+   * @param newBiDirectional This states wether the block is bidirectional
    */
-  public void setDirection(int newDirection) {
-    direction = newDirection;
+  public void setBiDirectional(Boolean newBiDirectional) {
+    biDirectional = newBiDirectional;

```

Comment:
you should pass "boolean newBidirectional", not "Boolean newBidirectional" - we want to use the primitive, not the wrapper class for boolean

Reply:
Also, it's usually convention to use the same parameter name as the instance variable:

public void setIsBidirectional (boolean isBidirectional) {
    this.isBidirectional = isBidirectional;
}

If you automatically generate the getters & setters then it will do this for you.